src/docbuilder/rustwide_builder.rs

```diff
@@ -383,6 +389,34 @@ impl RustwideBuilder {
                 (|| -> Result<bool> {
                     use docsrs_metadata::BuildTargets;
 
+                    let release_data = match self
+                        .index
+                        .api()
+                        .get_release_data(name, version)
+                        .context("error fetching release data from crates.io")
+                    {
+                        Ok(data) => data,
+                        Err(err) => {
+                            warn!("{:#?}", err);
+                            ReleaseData::default()
+                        }
+                    };
+
+                    if let Some(ref published_by) = release_data.published_by {
+                        info!(
+                            host_target_dir=?build.host_target_dir(),
+                            published_by_id=published_by.id,
+                            published_by_login=published_by.login,
+                            "restoring artifact cache",
+                        );
+                        if let Err(err) = self
+                            .artifact_cache
+                            .restore_to(&published_by.id.to_string(), build.host_target_dir())
+                        {
+                            warn!(?err, "could not restore artifact cache");
+                        }
+                    }
+
                     let mut has_docs = false;
                     let mut successful_targets = Vec::new();
                     let metadata = Metadata::from_crate_root(build.host_source_dir())?;
```

Comment:
nit: this whole closure is getting quite big, maybe we should separate it out into a function?

Reply:
This feels like I thing I would prefer to tackle together with #1011